### PR TITLE
added library/cloud/fleet module to control a remote CoreOS init system

### PR DESCRIPTION
**Pull request by coderfi**
I have been experimenting with CoreOS (which lacks a python interpreter) and was tired of typing manual fleet commands. So, I built this module in order to make it easier to construct Ansible playbooks to manage such clusters.
Hope others find this useful!

### CODE REVIEW

File: library/cloud/fleet [hunk 1]
```diff
@@ -0,0 +1,438 @@
...
+    description:
+      - Verify host keys presented by remote machines 
+        before initiating SSH connections.
+    required: true
+    default: 0
+    aliases: []
+
+# informational: requirements for nodes
+requirements: []
+author: Fairiz Azizi<coderfi@gmail.com>
```
Comment (resmo) [inline library/cloud/fleet]: guess, there is a missing space between your name and email address.

**Comment by coderfi** [comment]
oops, updated the author documentation (as well as a bug fix)

File: library/cloud/fleet [hunk 2]
```diff
@@ -0,0 +1,427 @@
+#!/usr/bin/python
+
+DOCUMENTATION = '''
+---
+module: fleet
+version_added: "0.5.0"
```
Comment (resmo) [inline library/cloud/fleet]: 0.5.0? this value is meant to be the version of ansible. 1.8 should be fine

**Comment by coderfi** [comment]
Oh, I thought that was the module version number.
Fixed.

**Comment by jwerak** [comment]
Hi, 
great job, thanks for this module.

did you try this module to list-machines against machines without tags? 

It fails for me when no tags are specified.

```
failed: [local] => {"failed": true, "parsed": false}
invalid output was: Traceback (most recent call last):
  File "/home/jveverka/.ansible/tmp/ansible-tmp-1411537815.71-190063171947348/fleet", line 1761, in <module>
    main()
  File "/home/jveverka/.ansible/tmp/ansible-tmp-1411537815.71-190063171947348/fleet", line 399, in main
    output)
  File "/home/jveverka/.ansible/tmp/ansible-tmp-1411537815.71-190063171947348/fleet", line 237, in parse_facts
    k, v = kv.split('=')
ValueError: need more than 1 value to unpack


FATAL: all hosts have already failed -- aborting
```

**Comment by coderfi** [comment]
Good catch. Should be fixed!

**Comment by jwerak** [comment]
cool, quick, thanks ;)

**Comment by mpdehaan** [comment]
Hi!

Thanks very much for your interest in Ansible.  It sincerely means a lot to us. 

On September 26, 2014, due to enormous levels of contribution to the project Ansible decided to reorganize module repos, making it easier
for developers to work on the project and for us to more easily manage new contributions and tickets.

We split modules from the main project off into two repos, http://github.com/ansible/ansible-modules-core and http://github.com/ansible/ansible-modules-extras

If you still would like this pull request merged, we will need your help making this target the new repo.  If you do not take any action, this
pull request unfortunately cannot be applied.

We apologize that we are not able to make this transition happen seamlessly, though this is a one-time change and your help is greatly appreciated -- 
this will greatly improve velocity going forward.

Both sets of modules will ship with Ansible, though they'll receive slightly different ticket handling.

To locate where a module lives between 'core' and 'extras'
- Find the module at http://docs.ansible.com/list_of_all_modules.html
- Open the documentation page for that module
- If the bottom of the docs say "This is an extras module", submit your ticket to https://github.com/ansible/ansible-modules-extras
- Otherwise, submit your pull request to update the existing module to https://github.com/ansible/ansible-modules-core
- Note that python modules in ansible now also end in ".py" and this extension is required for new contributions.
- action_plugins (modules with server side components) still live in the main repo.  If your pull request touches both, which should be
   exceedingly rare, submit two new pull requests and make sure to mention the links to each other in the comments.

Otherwise, if this is a new module:
- Submit your pull request to add a module to https://github.com/ansible/ansible-modules-extras

It may be possible to re-patriate your pull requests automatically, one user-submitted approach for advanced git users
has been suggested at https://gist.github.com/willthames/afbaaab0c9681ed45619

Additionally, should you need more help with this, you can ask questions on:
- the development mailing list: https://groups.google.com/forum/#!forum/ansible-devel

Thanks you very much!

**Comment by jwerak** [comment]
I've tried to list-units using this module and it didn't work. 

I've triggered the command created by fleet library and got following error: 
fleetctl -strict-host-key-checking='false' -tunnel=127.0.0.1:40002 -endpoint=http://127.0.0.1:4001 -known-hosts-file='~/.fleetctl/known_hosts'   list-units -full=true -fields=hash,unit,state,load,active,sub,machine,desc  
Invalid key in output format: "state"

fleetctl -strict-host-key-checking='false' -tunnel=127.0.0.1:40002 -endpoint=http://127.0.0.1:4001 -known-hosts-file='~/.fleetctl/known_hosts'   list-units -full=true -fields=hash,unit,load,active,sub,machine,desc
Invalid key in output format: "desc"

I am using fleet version from currently stable CoreOS 444.5.0:
fleetctl version 0.8.3

Do you have active pull request somewhere else?

**Comment by coderfi** [comment]
I'll take a look later tonight. It looks like those fields are not
supported in your fleet version. Should be an easy enough patch.

Fi

**Comment by coderfi** [comment]
Hi Vevrejak,

Sorry for the slow response.

I need to resubmit the pull request to ansible as they changed their
community pull repo.

In the meantime, you can get the code from

https://github.com/coderfi/ansible/blob/devel/library/cloud/fleet

All you need to do is copy the file to the same place in your local
directory.

As for the problem you have, I took a look and don't see a problem for
fleetctl v0.8.3 (I have the same)..

My guess is that you have two fleetctl versions running around.

The way that ansible and fleetctl would work is that the fleetctl used is
the one on your management machine (probably your local computer), and not
the remote one on the coreos host.

From where are you running fleetctl?

I typically run ansible/fleet from my MacBook PRO.
The latest fleetctl binaries are available from
https://github.com/coreos/fleet/releases

Please verify that 'fleetctl --version' shows v0.8.3 on your management
computer.

Hope this helps,
Fi

Fairiz "Fi" Azizi

**Comment by coderfi** [comment]
I stand corrected.... apparently I was the one who was using the wrong
fleetctl version (on my localhost). I was using v0.6.2 (I apparently copied
the latest version to the wrong directory so was accessing the old one
_grin_)

I will update the code to remove the desc and state fields since it is no
longer supported in v0.8.3

thanks for finding this!
Fi

Fairiz "Fi" Azizi

**Comment by coderfi** [comment]
Fixed.
I also moved it to a fork of the recently added ansible-modules-extras repo
https://github.com/coderfi/ansible-modules-extras/blob/devel/cloud/fleet.py

Here is the pull request

https://github.com/ansible/ansible-modules-extras/pull/91

Hope this helps!
Fi

Fairiz "Fi" Azizi

**Comment by jwerak** [comment]
Great, thanks :)

**Pull request closed without merging**